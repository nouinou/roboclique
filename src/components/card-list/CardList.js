import { Component } from 'react';
import Card from './card/Card';

class CardList extends Component {
	render() {
		const user = {
			id: '2',
			fullname: 'hamza nouinou',
			email: 'nouinouhamza@gmail.com',
			image: 'https://robohash.org/4?set=set3&size=192x192',
		};
		return (
			<div className="m-5 grid gap-5 xs:grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 justify-content-center">
				<Card user={user} />
				<Card user={user} />
				<Card user={user} />
				<Card user={user} />
				<Card user={user} />
				<Card user={user} />
			</div>
		);
	}
}

export default CardList;
