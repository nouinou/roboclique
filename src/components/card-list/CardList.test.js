import { shallow } from 'enzyme';
import CardList from './CardList';

describe('CardList Component', () => {
	const mockRobots = [
		{
			id: 1,
			fullname: 'Robo Cat',
		},
		{
			id: 2,
			fullname: 'Miaw Bot',
		},
	];
	it('should render CardList', () => {
		expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
	});
});
