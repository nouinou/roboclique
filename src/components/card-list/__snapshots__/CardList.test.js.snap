// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Component should render CardList 1`] = `
<div
  className="m-5 grid gap-5 xs:grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 justify-content-center"
>
  <Card
    key="1"
    robot={
      Object {
        "fullname": "Robo Cat",
        "id": 1,
      }
    }
  />
  <Card
    key="2"
    robot={
      Object {
        "fullname": "Miaw Bot",
        "id": 2,
      }
    }
  />
</div>
`;
